integration_tests/src/main/python/hive_delimited_text_test.py

```diff
@@ -270,24 +272,38 @@ def test_hive_text_round_trip_partitioned(spark_tmp_path, data_gen, spark_tmp_ta
     ArrayGen(string_gen),
     StructGen([('int_field', int_gen), ('string_field', string_gen)]),
     MapGen(StringGen(nullable=False), string_gen),
-    binary_gen
+    binary_gen,
+    StructGen([('b', byte_gen), ('i', int_gen), ('arr_of_i', ArrayGen(int_gen))]),
+    ArrayGen(StructGen([('b', byte_gen), ('i', int_gen), ('arr_of_i', ArrayGen(int_gen))]))
 ]
 
 
 @allow_non_gpu("org.apache.spark.sql.hive.execution.HiveTableScanExec")
 @pytest.mark.parametrize('data_gen', hive_text_unsupported_gens, ids=idfn)
 def test_hive_text_fallback_for_unsupported_types(spark_tmp_path, data_gen, spark_tmp_table_factory):
-    gen = StructGen([('my_field', data_gen)], nullable=False)
+    gen = StructGen([('my_supported_int_field', int_gen),
+                     ('my_unsupported_field', data_gen), ], nullable=False)
     data_path = spark_tmp_path + '/hive_text_table'
     table_name = spark_tmp_table_factory.get()
 
-    with_cpu_session(lambda spark: create_hive_text_table(spark, gen, table_name, data_path))
+    with_cpu_session(lambda spark: create_hive_text_table(spark,
+                                                          gen,
+                                                          table_name,
+                                                          data_path,
+                                                          "my_supported_int_field, my_unsupported_field"))
 
     assert_gpu_fallback_collect(
-            lambda spark: read_hive_text_table(spark, table_name),
+            lambda spark: read_hive_text_table(spark, table_name, "my_unsupported_field"),
             cpu_fallback_class_name=get_non_gpu_allowed()[0],
             conf=hive_text_enabled_conf)
 
+    # Even if the output-projection uses only supported types,
+    # the read should fall back to CPU if the table has unsupported types.
+    assert_gpu_fallback_collect(
```

Comment:
this is mostly for my own edification: we have to fall back in this scenario because the whole input is loaded to the gpu, so it is an all or nothing right? E.g. the GPU won't see an unsupported type and "skip it".

Reply:
That is correct. The whole file is consumed on GPU (i.e. all columns). The pruning of the output projection comes after the load, so we cannot do a partial read on GPU.

Reply:
P.S. That should have been clearer in the documentation. I'll add a note.